server/src/main/java/org/opensearch/common/settings/IndexScopedSettings.java

```diff
@@ -233,8 +236,7 @@ public final class IndexScopedSettings extends AbstractScopedSettings {
             IndexMetadata.INDEX_REMOTE_STORE_ENABLED_SETTING,
             IndexMetadata.INDEX_REMOTE_STORE_REPOSITORY_SETTING,
             IndexMetadata.INDEX_REMOTE_TRANSLOG_STORE_ENABLED_SETTING,
-            IndexMetadata.INDEX_REMOTE_TRANSLOG_REPOSITORY_SETTING,
-            IndexMetadata.INDEX_REMOTE_TRANSLOG_BUFFER_INTERVAL_SETTING
```

Comment:
Why is the setting removed from feature flagged settings and added to built-in?

Reply:
This was done to register the setting irrespective if remote store feature is enabled or not. This is harmless to register since the setting usage happens only when the remote translog is enabled. Also`INDEX_REMOTE_TRANSLOG_BUFFER_INTERVAL_SETTING.get(settings)` throws exception that setting is not registered. There is a workaround which was there earlier, but that was leading to null value being set in some of the tests while it should be set with some defaults when remote translog is enabled. Code ref -https://github.com/opensearch-project/OpenSearch/pull/8175/files#diff-2d3d2f64feacdc5d9fb97efe5966c878a3727554de568948a827ff2a91a6ffe2L778-R788